cpp/src/parquet/encoding.cc

```diff
@@ -3196,6 +3397,30 @@ class DeltaByteArrayDecoder : public DecoderImpl,
   std::shared_ptr<ResizableBuffer> buffered_data_;
 };
 
+class DeltaByteArrayDecoder : public DeltaByteArrayDecoderImpl<ByteArrayType> {
+ public:
+  using Base = DeltaByteArrayDecoderImpl<ByteArrayType>;
+  using Base::DeltaByteArrayDecoderImpl;
+
+  int Decode(ByteArray* buffer, int max_values) override {
+    return GetInternal(buffer, max_values);
+  }
+};
+
+class DeltaByteArrayFLBADecoder : public DeltaByteArrayDecoderImpl<FLBAType>,
+                                  virtual public FLBADecoder {
+ public:
+  using Base = DeltaByteArrayDecoderImpl<FLBAType>;
+  using Base::DeltaByteArrayDecoderImpl;
+
+  int Decode(ByteArray* buffer, int max_values) {
+    return GetInternal(buffer, max_values);
+  }
+  int Decode(FixedLenByteArray* buffer, int max_values) override {
```

Comment:
This is a bit confusing, why `DeltaByteArrayFLBADecoder` cannot decode FLBA?

Reply:
`GetInternal` only works with `ByteArray` at the moment. As serialization goes via ByteArray anyway I'm not sure if it's worth supporting an additional path. Maybe just cast instead?

Reply:
Yes, I think maybe check-and-cast is ok here, because `DeltaByteArrayFLBADecoder` cannot decode FLBA is really weird.